tests/Unit/EnclosureTest.php

```diff
@@ -88,6 +88,38 @@ public static function getLinkProvider(): iterable
             ,
             'http://example.net/link?a=%22b%22&amp;c=%3Cd%3E',
         ];
+
+        yield 'Test RSS 2.0 with channel link and enclosure' => [
+            <<<XML
+            <rss version="2.0" xmlns:media="http://search.yahoo.com/mrss/">
+                <channel>
+                    <link>http://example.net/tests/</link>
+                    <item>
+                        <link>/tests/3/</link>
+                        <media:content url="/images/3.jpg" medium="image"></media:content>
+                    </item>
+                </channel>
+            </rss>
+XML
+            ,
+            'http://example.net/images/3.jpg',
+        ];
```

Comment:
This was the original bug I faced, which had me investigate the issue (which turned out to be more severe and complex than anticipated...)